src/main/webapp/app/exercises/programming/shared/instructions-render/programming-exercise-instruction.component.ts

```diff
@@ -96,6 +96,16 @@ export class ProgrammingExerciseInstructionComponent implements OnChanges, OnIni
      * @param changes
      */
     public ngOnChanges(changes: SimpleChanges) {
+        if (this.exercise?.isAtLeastTutor) {
+            this.testCasesSubscription = this.programmingExerciseGradingService
```

Comment:
This opens a new subscription on changes, but does not destroy the old one if it's still present. Can you add a check and unsubscribe if necessary.